pkg/controller/migplan/migplan_controller.go

```diff
@@ -276,6 +283,81 @@ func (r *ReconcileMigPlan) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+// Ensure refresh is performed if requested
+func (r *ReconcileMigPlan) ensureRefresh(plan *migapi.MigPlan) error {
+	// A "Refresh", triggered by `plan.spec.refresh = true` will reconcile
+	// the MigPlan, MigStorage, and MigClusters.
+	// When all of these items are reconciled, `spec.refresh` on each
+	// will be set back to `false`
+
+	var srcCluster, dstCluster *migapi.MigCluster
+	var storage *migapi.MigStorage
+	var err error
+
+	// Abort if items to be refreshed are not ready
+	if plan.Status.HasAnyCondition(
+		SourceClusterNotReady, InvalidSourceClusterRef,
+		DestinationClusterNotReady, InvalidDestinationClusterRef,
+		StorageNotReady, InvalidStorageRef,
+	) {
+		plan.Spec.Refresh = false
```

Comment:
In the case that I can't perform the refresh due to bad k8s object refs etc., I'm giving up on it.